solana/programs/wormhole-governance/src/instructions/governance.rs

```diff
@@ -71,14 +75,223 @@ pub struct Governance<'info> {
+                "Invalid GovernanceMessage chain",
+            ));
+        }
+
+        let program_id: Pubkey = Pubkey::new_from_array(Readable::read(reader)?);
+        let accounts_len: u16 = Readable::read(reader)?;
+        let mut accounts = Vec::with_capacity(accounts_len as usize);
+        for _ in 0..accounts_len {
+            let pubkey: [u8; 32] = Readable::read(reader)?;
+            let is_signer: bool = Readable::read(reader)?;
+            let is_writable: bool = Readable::read(reader)?;
+            accounts.push(Acc {
+                pubkey: Pubkey::new_from_array(pubkey),
+                is_signer,
+                is_writable,
+            });
+        }
+        let data_len: u16 = Readable::read(reader)?;
+        let mut data = vec![0; data_len as usize];
+        reader.read_exact(&mut data)?;
+
+        Ok(GovernanceMessage {
+            program_id,
+            accounts,
+            data,
+        })
+    }
+}
+
+impl Writeable for GovernanceMessage {
+    fn written_size(&self) -> usize {
+        Self::MODULE.len()
+        + GovernanceAction::SIZE.unwrap() // action
+        + u16::SIZE.unwrap() // chain
+        + <[u8; 32]>::SIZE.unwrap() // program_id
+        + u16::SIZE.unwrap() // accounts_len
+        + self.accounts.iter()  // accounts
+          .map(|_| 32 + 1 + 1).sum::<usize>()
+        + u16::SIZE.unwrap() // data_len
+        + self.data.len() // data
+    }
+
+    fn write<W>(&self, writer: &mut W) -> io::Result<()>
+    where
+        W: io::Write,
+    {
+        let GovernanceMessage {
+            program_id,
+            accounts,
+            data,
+        } = self;
+
+        Self::MODULE.write(writer)?;
+        GovernanceAction::SolanaInstruction.write(writer)?;
+        u16::from(Chain::Solana).write(writer)?;
+        program_id.to_bytes().write(writer)?;
+        (accounts.len() as u16).write(writer)?;
+        for acc in accounts {
+            acc.pubkey.to_bytes().write(writer)?;
+            acc.is_signer.write(writer)?;
+            acc.is_writable.write(writer)?;
+        }
+        (data.len() as u16).write(writer)?;
+        writer.write_all(data)
+    }
+}
+
+#[test]
+fn test_governance_message_serde() {
+    let program_id = Pubkey::new_unique();
+    let accounts = vec![
+        Acc {
+            pubkey: Pubkey::new_unique(),
+            is_signer: true,
+            is_writable: true,
+        },
+        Acc {
+            pubkey: Pubkey::new_unique(),
+            is_signer: false,
+            is_writable: true,
+        },
+    ];
+    let data = vec![1, 2, 3, 4, 5];
+    let msg = GovernanceMessage {
+        program_id,
+        accounts,
+        data,
+    };
+
+    let mut buf = Vec::new();
+    msg.serialize(&mut buf).unwrap();
+
+    let msg2 = GovernanceMessage::deserialize(&mut buf.as_slice()).unwrap();
+    assert_eq!(msg, msg2);
+}
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum GovernanceAction {
+    Undefined,
+    SolanaInstruction,
```

Comment:
This enum would seem to conflict with the one defined here https://github.com/wormhole-foundation/example-native-token-transfers/blob/main/evm/src/wormhole/Governance.sol#L14-L17